chain/chain/src/store/mod.rs

```diff
@@ -282,6 +283,93 @@ pub struct ChainStore {
     save_trie_changes: bool,
     /// The maximum number of blocks for which a transaction is valid since its creation.
     pub(super) transaction_validity_period: BlockHeightDelta,
+    /// Guarantueed to be updated when storage is updated, so can be considered a source of truth.
+    cache: ChainStoreCache,
+}
+
+#[derive(Default)]
+pub struct ChainStoreCache {
+    // TODO max_distance: u64
+
+    // TODO Consider which container to use.
+    // Interior mutability will not be required anymore once the apply-range command takes
+    // care of populating the cache. See comments below regarding apply-range not calling finalize.
```

Comment:
Hm, yeah, `apply-range` can't `finalize` as that would advance the chain, which is in direct opposition if what we need it to do when benchmarking. I'd say lets add a method that just fills the caches that gets called from `finalize` and then separately by `apply-range` if its in benchmark mode.